antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/tree/xpath/XPathLexer.kt

```diff
@@ -26,44 +27,35 @@ public open class XPathLexer(public val input: CharStream) : Lexer(input) {
 
     private val SHARED_CONTEXT_CACHE = PredictionContextCache()
 
-    private val LITERAL_NAMES: Array<String?> =
-      arrayOf(null, null, null, "'//'", "'/'", "'*'", "'!'")
+    private val LITERAL_NAMES: Array<String?> = arrayOf(
+      null, null, null, "'//'", "'/'", "'*'", "'!'"
+    )
 
-    private val SYMBOLIC_NAMES: Array<String?> =
-      arrayOf(null, "TokenRef", "RuleRef", "Anywhere", "Root", "Wildcard", "Bang", "Id", "String")
+    private val SYMBOLIC_NAMES: Array<String?> = arrayOf(
+      null, "TokenRef", "RuleRef", "Anywhere", "Root", "Wildcard", "Bang", "Id", "String"
+    )
 
     private val VOCABULARY = VocabularyImpl(LITERAL_NAMES, SYMBOLIC_NAMES)
   }
 
-  public enum class Tokens(public val id: Int) {
-    TokenRef(1),
-    RuleRef(2),
-    Anywhere(3),
-    Root(4),
-    Wildcard(5),
-    Bang(6),
-    Id(7),
-    String(8),
-  }
-
-  public enum class Channels(public val id: Int) {
-    DEFAULT_TOKEN_CHANNEL(0),
-    HIDDEN(1),
+  public object Tokens {
```

Comment:
Isn't an Enum preferable, as the compiler can check we are covering all cases in when statements

Reply:
Forgot to mention that Enums where not really *used*. The way the original code worked was just wrapping an `id` (the numeric value) inside of the enum entry, but the enums itself where never used directly, so this statement

> as the compiler can check we are covering all cases in when statements

wasn't true even before.